addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -14,6 +14,7 @@ const _t = core._t;
 const MESSAGE_FETCH_LIMIT = 30;
 const PREVIEW_MSG_MAX_SIZE = 350;
 
+// FIXME {xdu} maybe use something like a function generator ?
```

Comment:
maybe use `_uniqueId()` with clear naming convention for temporary, e.g. `_uniqueId('mail.temporary*_')`? That way, id part is handled by underscorejs, but "model" part clearly tells they are temporary.

Reply:
Do I apply this on all temporaries ? (for Message, Attachment and Thread ?)
We do not care about any order in temporaries ? I know for thread it's not useful and for attachment neither I think but for message I don't know ?

Reply:
Hm... That's a good remark.
The order should be based on id, but this doesn't really work when mixing temporaries and actuals.

ATM, temporaries are either not used with actuals (messages & threads), or treated as short-lived and ordered based on creation (attachments), so we don't have the issue.

I haven't given much thoughts about desiring having a well-defined ordered list of (temporary & actual) entities. I feel like the wording "temporary" is appropriate for attachment, but not for message and thread.

Reply:
Indeed, they are more "faky" or "notStoredInDatabase" ones...

Reply:
I created a ticket for that : https://www.odoo.com/web#id=2166386&action=3206&active_id=1519&model=project.task&view_type=form&menu_id=4720